pygpu/tools.py

```diff
@@ -127,15 +127,16 @@ def check_args(args, collapse=False, broadcast=False):
     if dims is None:
         raise TypeError("No arrays in kernel arguments, "
                         "something is wrong")
-    tdims = dims
 
     if broadcast or collapse:
         # make the strides and dims editable
         dims = list(dims)
         strs = [list(str) if str is not None else str for str in strs]
 
     if broadcast:
+        # Set strides to 0s when needed.
         # Get the full shape in dims (no ones unless all arrays have it).
+        tdims = dims
```

Comment:
tdims is only used in this if. So it will always be defined in the part needed. I'll move it back to have it stay a tuple.